client/services/SolutionHandlerService.js

```diff
@@ -45,19 +45,28 @@ tie.factory('SolutionHandlerService', [
           // Otherwise, the code doesn't have any obvious syntax errors.
           // Generate a CodeSubmission object that wraps the student's code
           // into a class and appends some test code, then run the whole thing.
+          var correctnessTests = [];
+          var buggyOutputTests = [];
+          var performanceTests = [];
+          for (var i = 0; i < tasks.length; i++) {
+            correctnessTests.push(tasks[i].getCorrectnessTests());
+            buggyOutputTests.push(tasks[i].getBuggyOutputTests());
+            performanceTests.push(tasks[i].getPerformanceTests());
+          }
+          var lastTask = tasks[tasks.length - 1];
           var codeSubmission = CodeSubmissionObjectFactory.create(
             studentCode.trim());
           CodePreprocessorDispatcherService.preprocess(
             language, codeSubmission, auxiliaryCode,
-            task.getInputFunctionName(), task.getMainFunctionName(),
-            task.getOutputFunctionName(), task.getCorrectnessTests(),
-            task.getBuggyOutputTests(), task.getPerformanceTests());
+            lastTask.getInputFunctionName(), lastTask.getMainFunctionName(),
```

Comment:
In the internationalization question, there are two tasks, which involve the student writing two separate functions. The first task contains one function name, and the second task contains a different function name. (The second task's function will make a call to the first task's function.)

When doing the test run that involves both task 1 and 2, the first function name should still be used when evaluating correctness for the first task. It sounds like you're not doing that?